pkg/discovery/network/flannel.go

```diff
@@ -28,65 +28,68 @@ import (
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	controllerClient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 //nolint:nilnil // Intentional as the purpose is to discover.
 func discoverFlannelNetwork(ctx context.Context, client controllerClient.Client) (*ClusterNetwork, error) {
 	daemonsets := &appsv1.DaemonSetList{}
 
-	err := client.List(ctx, daemonsets, controllerClient.InNamespace(metav1.NamespaceSystem))
+	// Use a label selector to find DaemonSets with the "k8s-app=flannel" label in any namespace
+	labelSelector := controllerClient.MatchingLabels{"k8s-app": "flannel"}
+	err := client.List(ctx, daemonsets, labelSelector)
```

Comment:
Maybe you can use controllerClient.MatchingLabels{"k8s-app": "flannel"} directly in List function

Reply:
Make sense, made the change